webrtc.html

```diff
@@ -8942,6 +8956,54 @@ <h2 id="sec.stats-model">Statistics Model</h2>
       [[!WEBRTC-STATS]].</p>
     </section>
     <section>
+      <h3>The stats selection algorithm</h3>
+      <p>
+        The <dfn>stats selection algorithm</dfn> is as follows:
+        <ol>
+          <li>
+            Let <var>result</var> be an empty RTCStatsReport.
+          </li>
+          <li>If <var>selector</var> is <code>null</code>, gather stats for the
+            whole <var>connection</var>, add them to <var>result</var>, return
+            <var>result</var>, and abort these steps.
+          </li>
+          <li>If <var>selector</var> is an <a>RTCRtpSender</a>, gather stats for
+            and add the following objects to <var>result</var>:
+            <ul>
+              <li>
+              All <code>RTCInboundRTPStreamStats</code> objects corresponding to
+              <var>selector</var>.
+              </li>
+              <li>
+              All stats objects referenced directly or indirectly by the
+              <code>RTCInboundRTPStreamStats</code> objects added.
+              </li>
+            </ul>
+          </li>
+          <li>If <var>selector</var> is an <a>RTCRtpReceiver</a>, gather stats
+            for and add the following objects to <var>result</var>:
+            <ul>
+              <li>
+              All <code>RTCOutboundRTPStreamStats</code> objects corresponding
+              to <var>selector</var>.
+              </li>
+              <li>
+              All stats objects referenced directly or indirectly by the
+              <code>RTCOutboundRTPStreamStats</code> added.
+              </li>
+            </ul>
+          </li>
```

Comment:
I believe this pulls in stats dictionaries of all types except for `RTCMediaStreamStats` (of which the track belongs), `RTCDataChannelStats` and `RTCPeerConnectionStats`. But only a subset of all object of each type need to be collected.

In terms of avoiding unnecessary collection of stats for performance, I'm still worried about the complexity of implementing this. We need to resolve references first, and collect stats afterwards, and only collect for relevant objects (otherwise there's no point to the selector). This should be doable, my concern stems with (Chromium's WebRTC codebase's) implementation details; lower layer objects not perfectly mapping with the spec, how references between objects are resolved, threads and collection order when it's not as pretty of a tree as I would like.

But based on my whiteboard sketching this should be doable.